### ISSUE TITLE
A little suggestion: use "import" to load the js scripts

### ISSUE THREAD

**Issue by yueyinqiu** [magiccodingman/Magic.IndexedDb#1]
An amazing project! Actually, this might be the best IndexedDb library for blazor I have ever found. Most libraries depend on `TG.Blazor.IndexedDB`, which has no update for 3 years and is still in preview. And this project is also very young, which means it might be much more easy to have breaking changes currently. So I feel very exciting to find the project at this time.

And here is my first suggestion, just a little suggestion: using "import" to load the js scripts.

Currently it's running `window.magicBlazorDB = ...` in js in advance, and then using `_jsRuntime.InvokeAsync<TResult>("window.magicBlazorDB.xxxxxxx")` in c# to invoke the functions. This method could work, but it requires the users to manually write `<script src="_content/Magic.IndexedDb/magicDB.js"></script>`. More importantly, it will occupy the property `window.magicBlazorDB`. Although hardly another packages will also use this property, it is a hidden trouble.

Instead, using "import" could avoid the above problem. For this project, we can use `var module = await JS.InvokeAsync<IJSObjectReference>("import", "./_content/Magic.IndexedDb/magicDB.js")` to get the module, and then use `module.InvokeAsync<TResult>("xxxxx")` to invoke the functions. Of course, `magicDB.js` should be rewrited as a module at the same time, like:

```javascript
import ... from "./dexie.min.js"; // also directly import dexie here
export function createDb (dotnetReference, transaction, dbStore) {
    ...
}
export function ...
...
```

In this way, it don't need to assign any value to `window.magicBlazorDB`, and users can get rid of writing the lines like `<script src="_content/Magic.IndexedDb/magicDB.js"></script>`. I think this could make the package better. Finally, thanks for your great work.

**Comment by yueyinqiu** [comment]
another suggestion now is to deal with the nullable reference.... 102 warnings about it.... this used to be written in lower dotnet versions?

**Comment by magiccodingman** [comment]
> another suggestion now is to deal with the nullable reference.... 102 warnings about it.... this used to be written in lower dotnet versions?

I just pushed an update (not to nuget package yet) that fixes all 60+ null reference warnings and I also accidentally broke the AddRange last update and just repaired that as well.

**Comment by yueyinqiu** [comment]
Haha, you're much better than me. In fact I have also tried to write a IndexedDb package, and my aim was just to make a simple wrapper for it but still failed. It's really a hard work.

And about dexie.js, my suggestion is to regard it as an implementation detail.

Comparing to `dropping off the planet -> no more update -> users can manually choose the new versions of dexie.js`, I'm much more worried about that if there is a breaking change of dexie.js, and this chain will become `users are choosing the new versions of dexie.js -> undefined, error, ... -> what a shit package (╯‵□′)╯` even if the package is still updating but just a bit later than dexie.js (and it must be).

Meanwhile, for most users, there are no reasons for them to specify the version of dexie.js. If you are afraid of `dropping off the planet`, just believe that there will be someone continuing your great work! ;)

**Comment by magiccodingman** [comment]
@yueyinqiu You're 100% right. Allowing different versions of dexie not part of the package is more likely to create breaking changes since Dexie is still updated consistently. So it's likely they depreciate code as they've already done in the past as I've skimmed through much of their documentation and I've seen that before. So, you've convinced me! I appreciate the suggestions/advice. I'm all ears. I'll be adding the import hopefully tonight when I have a bit to sit down. 

I've been adding to this package as I run into things since I built this to accomplish required goals I have on a separate project. So as my side project project moves alone, I'm dragging this project with it :) I've been building out some cool SignalR to API sync functionalities and stuff like that as well. I'm honestly really glad someone took a look at the code. It was a really fun project that I wanted to talk to others about it :D Thank you!
